packages/compartment-mapper/src/bundle-cjs.js

```diff
@@ -14,15 +16,28 @@ const exportsCellRecord = exportsList =>
 const runtime = function wrapCjsFunctor(num) {
   /* eslint-disable no-undef */
   return ({ imports = {} }) => {
+    const moduleCells = cells[num];
     const cModule = Object.freeze(
-      Object.defineProperty({}, 'exports', cells[num].default),
+      Object.defineProperty({}, 'exports', moduleCells.default),
     );
     // TODO: specifier not found handling
     const requireImpl = specifier => cells[imports[specifier]].default.get();
     functors[num](Object.freeze(requireImpl), cModule.exports, cModule);
-    Object.keys(cells[num])
+    // Update all named cells from module.exports.
+    Object.keys(moduleCells)
       .filter(k => k !== 'default' && k !== '*')
-      .map(k => cells[num][k].set(cModule.exports[k]));
+      .map(k => moduleCells[k].set(cModule.exports[k]));
+    // Add new named cells from module.exports.
+    Object.keys(cModule.exports)
+      .filter(k => k !== 'default' && k !== '*')
+      .filter(k => moduleCells[k] === undefined)
+      .map(k => (moduleCells[k] = cell(k, cModule.exports[k])));
+    // Update the star cell from all cells.
```

Comment:
I mentioned earlier that in regular endo apps the `*` behavior is implemented elsewhere, but that elsewhere doesn't exist in bundler, so I think this is the right place to do this.